dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/CDIContainerImpl.java

```diff
@@ -652,6 +652,25 @@ private ExtensionArchive newSPIExtensionArchive(ServiceReference<CDIExtensionMet
             applicationBDAsVisible = internalExtension.applicationBeansVisible();
         }
 
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "newSPIExtensionArchive", "***We are creating a new CDI Extension Archive***");
+            if (! extra_classes.isEmpty()) {
+                Tr.debug(tc, "newSPIExtensionArchive", "The following classes will be registered as beans: " + String.join(", ", extra_classes));
+            }
+
+            if (! extensionClassNames.isEmpty()) {
+                Tr.debug(tc, "newSPIExtensionArchive", "The following classes will be registered as extensions: " + String.join(", ", extensionClassNames));
+            }
+
+            if (extraAnnotations.isEmpty()) {
+                Tr.debug(tc, "newSPIExtensionArchive", "The following annotations will be registered as bean defining annotations: " + String.join(", ", extraAnnotations));
+            }
```

Comment:
This needs a `!`

Reply:
Actually, it might be better to remove the `if` altogether, so that the trace shows explicitly when there are no extra annotations or extra class names etc.